prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -853,6 +886,18 @@ ssize_t rxr_pkt_proc_matched_rtm(struct rxr_ep *ep,
 		rx_entry->cq_entry.len = rx_entry->total_len;
 
 	pkt_type = rxr_get_base_hdr(pkt_entry->pkt)->type;
+
+	if (pkt_type >= RXR_DC_REQ_PKT_BEGIN &&
+	    pkt_type <= RXR_BASELINE_DC_REQ_PKT_END)
+		rx_entry->fi_flags |= RXR_DELIVERY_COMPLETE_REQUESTED;
+
+	if (pkt_type == RXR_LONG_MSGRTM_PKT || pkt_type == RXR_LONG_TAGRTM_PKT)
+		rx_entry->tx_id = rxr_get_long_rtm_base_hdr(pkt_entry->pkt)->tx_id;
+	else
```

Comment:
add ```if (pkt_type == RXR_EAGER_MSGRTM_PKT || pkt_type == RXR_EAGER_TAGRTM_PKT)``` here. Because for regular eager message packet, you cannot call ```get_dc_eager_rtm_base_hdr```, They does not need ```tx_id``` either.

Reply:
Ok. I will update it.

Reply:
Current logic does not set `rx_entry->tx_id` for `RXR_EAGER_MSGRTM_PKT` and `RXR_EAGER_TAGRTM_PKT` pkt types, https://github.com/ofiwg/libfabric/pull/6205/files#diff-406e99ade0d03e770540fc56cc444ed448d046cd3065b645f54f791e05c7b6f9R988